src/main/java/org/eclipse/yasson/internal/components/JsonbComponentInstanceCreator.java

```diff
@@ -15,13 +15,21 @@
 
 import java.io.Closeable;
 
+
 /**
  * Creates instances of JsonbComponents such as JsonbAdapter.
- * If CDI is available uses BeanManager to create instance, otherwise calls no parameter constructor.
- *
+ * 
+ * <p>
+ * Yasson attempts to load the implementations using {@link java.util.ServiceLoader} first. If there are multiple
+ * implementations found the service provider with the highest priority is used. If there are no implementations found the
+ * {@link JsonbComponentInstanceCreatorFactory} is used to instantiate the default implementation.
+ * </p>
+ * 
  * @author Roman Grigoriadi
  */
 public interface JsonbComponentInstanceCreator extends Closeable {
```

Comment:
lets move this interface to a new `org.eclipse.yasson.spi` package and export it in the module-info.java. If we left this interface here, other libraries would not be able to actually implement this interface under the constraints of JPMS.

Reply:
Ok. To be honest I did not consider JPMS at all.